app/models/api_key_rubygem_scope.rb

```diff
@@ -0,0 +1,13 @@
+class ApiKeyRubygemScope < ApplicationRecord
+  belongs_to :api_key
+  belongs_to :ownership
+  validates :ownership_id, uniqueness: { scope: :api_key_id }
+  validates :api_key, :ownership, presence: true
+  before_destroy :soft_delete_api_key!, if: :destroyed_by_association
```

Comment:
Without the `if: destroyed_by_association`, changing a gem scope from a gem to all gems calls `soft_delete_api_key!`